### ISSUE TITLE
Apple M1 building instructions

### ISSUE THREAD

**Issue by fabiosirna** [darktable-org/darktable#14656]
**Is your feature request related to a problem? Please describe.**

I'd like to build myself darktable from the sources, but all the instructions I've found are not working.

**Describe the solution you'd like**

A wiki page with all the steps to build the package using homwbrew or macports

**Alternatives**

I tried to build the  source using the current instructions, and trying to troubleshooting also with ChatGPT help. But I'm not a developer and it's very hard to build without some knowledge.

**Comment by fabiosirna** [comment]
If can help, I'm attaching here the latest logs.

**Comment by zisoft** [comment]
With what package manager are you trying to build? Homebrew or MacPorts.

And yes, a log would help.

**Comment by fabiosirna** [comment]
My previous comment for some reason doesn't contains the logs.

I'm currently following the instructions for [ARM64 processor](https://raw.githubusercontent.com/darktable-org/darktable/master/packaging/macosx/BUILD-ARM64.txt) using Macports. 

I followed all the steps and after typing '''make''' the process stop around 42% with the following details:

```bash
[ 42%] Linking C shared library libdarktable.dylib
undef: _u_errorName_73
undef: _u_foldCase_73
undef: _u_strToLower_73
undef: _u_strToUpper_73
undef: _ucol_close_73
undef: _ucol_open_73
undef: _ucol_strcoll_73
undef: _uregex_close_73
undef: _uregex_matches_73
undef: _uregex_open_73
undef: _uregex_setText_73
Undefined symbols for architecture arm64:
  "_u_errorName_73", referenced from:
      _icuLoadCollation in sqliteicu.c.o
      _icuRegexpFunc in sqliteicu.c.o
      _icuCaseFunc16 in sqliteicu.c.o
  "_u_foldCase_73", referenced from:
      _icuLikeCompare in sqliteicu.c.o
  "_u_strToLower_73", referenced from:
      _icuCaseFunc16 in sqliteicu.c.o
  "_u_strToUpper_73", referenced from:
      _icuCaseFunc16 in sqliteicu.c.o
  "_ucol_close_73", referenced from:
      _icuLoadCollation in sqliteicu.c.o
      _icuCollationDel in sqliteicu.c.o
  "_ucol_open_73", referenced from:
      _icuLoadCollation in sqliteicu.c.o
  "_ucol_strcoll_73", referenced from:
      _icuCollationColl in sqliteicu.c.o
  "_uregex_close_73", referenced from:
      _icuRegexpDelete in sqliteicu.c.o
  "_uregex_matches_73", referenced from:
      _icuRegexpFunc in sqliteicu.c.o
  "_uregex_open_73", referenced from:
      _icuRegexpFunc in sqliteicu.c.o
  "_uregex_setText_73", referenced from:
      _icuRegexpFunc in sqliteicu.c.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/libdarktable.dylib] Error 1
make[1]: *** [bin/CMakeFiles/lib_darktable.dir/all] Error 2
make: *** [all] Error 2

```

Here below the logs:

[CMakeError.log](https://github.com/darktable-org/darktable/files/11668700/CMakeError.log)
[CMakeOutput.log](https://github.com/darktable-org/darktable/files/11668701/CMakeOutput.log)

**Comment by fabiosirna** [comment]
Driven by curiosity I also tried the [homebrew method](https://github.com/darktable-org/darktable/blob/master/packaging/macosx/BUILD_hb.txt) and I got a similar error for libdarktable.dylib after launching the script at point 2 option A `2_build_hb_darktable_default.sh` :

```bash
[ 54%] Linking C shared library libdarktable.dylib
Undefined symbols for architecture arm64:
  "_u_errorName_73", referenced from:
      _icuLoadCollation in sqliteicu.c.o
      _icuRegexpFunc in sqliteicu.c.o
      _icuCaseFunc16 in sqliteicu.c.o
  "_u_foldCase_73", referenced from:
      _icuLikeCompare in sqliteicu.c.o
  "_u_strToLower_73", referenced from:
      _icuCaseFunc16 in sqliteicu.c.o
  "_u_strToUpper_73", referenced from:
      _icuCaseFunc16 in sqliteicu.c.o
  "_ucol_close_73", referenced from:
      _icuLoadCollation in sqliteicu.c.o
      _icuCollationDel in sqliteicu.c.o
  "_ucol_open_73", referenced from:
      _icuLoadCollation in sqliteicu.c.o
  "_ucol_strcoll_73", referenced from:
      _icuCollationColl in sqliteicu.c.o
  "_uregex_close_73", referenced from:
      _icuRegexpDelete in sqliteicu.c.o
  "_uregex_matches_73", referenced from:
      _icuRegexpFunc in sqliteicu.c.o
  "_uregex_open_73", referenced from:
      _icuRegexpFunc in sqliteicu.c.o
  "_uregex_setText_73", referenced from:
      _icuRegexpFunc in sqliteicu.c.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [bin/CMakeFiles/lib_darktable.dir/build.make:3384: bin/libdarktable.dylib] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2496: bin/CMakeFiles/lib_darktable.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
./2_build_hb_darktable_default.sh{}:26: Error: command `../../build.sh --install --build-type Release --prefix "$installDir"` failed with exit code 2

```

**Comment by ralfbrown** [comment]
Looks like you're missing libicu, even though you have the headers for it installed.  My guess is that you installed the -dev package but it doesn't list the package containing the acutal library as a dependency.

**Comment by fabiosirna** [comment]
@ralfbrown library is installed here: `/opt/homebrew/opt/icu4c/include/unicode`

I'm trying to reinstall it: `brew reinstall icu4c`

If doesn't work, I'll try with a fresh reinstall of homebrew 🤞🏻

**Comment by fabiosirna** [comment]
I made some progress.

I checked my homebrew setup, run` brew doctor` to fix some issues.
Then, referring to [these instructions](https://github.com/darktable-org/darktable/blob/master/packaging/macosx/BUILD_hb.txt), I was able to run both:
`1_install_hb_dependencies.sh`
`2_build_hb_darktable_default.sh`

without any error.

As soon as I run the script to build the package:
```3_make_hb_darktable_package.sh```

I get:

```
Installing dependency </opt/homebrew/opt/pugixml/lib/libpugixml.1.dylib> of <../../build/macosx/package/darktable.app/Contents/MacOS/darktable>.
dyld[4514]: Symbol not found: _g_task_set_static_name
  Referenced from: <FD8A75C2-18CE-35DD-9644-D8429D7AB1A4> /opt/homebrew/Cellar/gdk-pixbuf/2.42.10_1/lib/libgdk_pixbuf-2.0.0.dylib
  Expected in:     <66492145-2AFC-3816-BF25-333BFDB23084> /opt/homebrew/Cellar/glib/2.74.5/lib/libgio-2.0.0.dylib
packaging/macosx/3_make_hb_darktable_package.sh: line 220:  4514 Abort trap: 6           gtk3-update-icon-cache -f "$dtResourcesDir"/share/icons/Adwaita
packaging/macosx/3_make_hb_darktable_package.sh{}:220: Error: command `gtk3-update-icon-cache -f "$dtResourcesDir"/share/icons/Adwaita` failed with exit code 134
```

I've checked the full path and run manually the command:
`gtk3-update-icon-cache -f /Users/fabio/Developer/darktable/build/macosx/package/darktable.app/Contents/Resources/share/icons/Adwaita`

and I get the same error above.

It looks like that the symbol `_g_task_set_static_name` is missing in the `libgdk_pixbuf-2.0.0.dylib` library or is incompatible with the current version of the `libgio-2.0.0.dylib` library, and I don't know how to check/fix.

Thanks for any suggestions! Looks like I'm almost there :)

**Comment by zisoft** [comment]
[glib is at 2.76.3 on homebrew](https://formulae.brew.sh/formula/glib), can you try to update?

**Comment by fabiosirna** [comment]
I managed to build darktable on my Macbook M1 :)

```
brew uninstall --ignore-dependencies gdk-pixbuf glib
brew install glib gdk-pixbuf protobuf
```

it worked to build the source.

Then, when I was trying to build the packaged I got another error related to libtiff missing. I installed manually and aftere re-run the build with `./3_make_hb_darktable_package.sh` I had to fake the missing library copy the version 5 to 6.

```
brew install libtiff
cp /opt/homebrew/opt/libtiff/lib/libtiff.5.dylib /opt/homebrew/opt/libtiff/lib/libtiff.6.dylib
```
libtiff is missing in the `1_install_hb_dependencies.sh` script.

In the [build notes](https://github.com/darktable-org/darktable/blob/master/packaging/macosx/BUILD_hb.txt) a disclaimer to check homebrew with `brew doctor` BEFORE trying to run the scripts will be a good reminder for those who have homebrew already installed in their system.

Attaching proof :)

![Screenshot 2023-06-07 at 09 21 28](https://github.com/darktable-org/darktable/assets/2249234/2f9ce174-dabd-4801-909b-065996394ffd)
